components/datetime/src/error.rs

```diff
@@ -38,6 +39,9 @@ pub enum DateTimeFormatError {
     /// An error originating from [`DateTimeInput`][icu_datetime::date::DateTimeInput].
     #[displaydoc("{0}")]
     DateTimeInput(DateTimeError),
+    /// An error originating from a missing symbol in the data.
+    #[displaydoc("Data file missing {0} symbol for key {1}")]
+    MissingSymbol(&'static str, String),
```

Comment:
Suggestion: Make different error kinds for each field (weekday, month, and era).  Advantages:

1. More type-safe; avoids need to programmatically check what is in the `&'static str`
2. Allows for the key to be an integer for weekday and month and a tinystr for era, rather than pulling in the String machinery for an error case